clang/lib/Basic/Targets/AArch64.cpp

```diff
@@ -197,6 +200,29 @@ AArch64TargetInfo::AArch64TargetInfo(const llvm::Triple &Triple,
         Opts.EABIVersion == llvm::EABI::GNU ? "\01_mcount" : "mcount";
 }
 
+void AArch64TargetInfo::validatePAuthOptions(DiagnosticsEngine &Diags,
+                                             const LangOptions &Opts) const {
+  if (Opts.SoftPointerAuth)
+    return;
+
+  bool NeedsFeatPAuth = false;
+  // FIXME: Some ptrauth_* intrinsics can be implemented without FEAT_PAuth.
```

Comment:
Is this still true even with `Opts.SoftPointerAuth == false`? Isn't the if statement above enough?

Reply:
This comment is about operations like sign and auth using IA or IB keys - these can be performed with only HINT-encoded instructions (but this is not the case for DA and DB keys). I guess these may be useful in some run-time libraries. (If they are definitely not useful, it may be worth commenting llvm/llvm-project#74729 and llvm/llvm-project#72502 as well)

Reply:
Fixed